app/controllers/components_controller.rb

```diff
@@ -12,7 +12,11 @@ def show
   end
 
   def create
-    @component = NewComponentEvent.create(component_params)
+    @component = if params.dig('component', 'component_type') == 'MSA'
```

Comment:
Could you have not used a ternary operator here?

Reply:
Wrote it out and then felt it was a little too long 😆

Reply:
Fair enough :-)